src/core/Akka.Remote/Endpoint.cs

```diff
@@ -185,26 +187,29 @@ public ShutDownAssociationException(Address localAddress, Address remoteAddress,
         public Address RemoteAddress { get; private set; }
     }
 
-    internal sealed class InvalidAddressAssociationException : EndpointException, IAssociationProblem
+    internal sealed class InvalidAssociation : EndpointException, IAssociationProblem
     {
-        public InvalidAddressAssociationException(Address localAddress, Address remoteAddress, Exception cause = null)
+        public InvalidAssociation(Address localAddress, Address remoteAddress, Exception cause = null, DisassociateInfo? disassociateInfo = null)
             : base(string.Format("Invalid address: {0}", remoteAddress), cause)
         {
             RemoteAddress = remoteAddress;
             LocalAddress = localAddress;
+            DisassociationInfo = disassociateInfo;
         }
 
         public Address LocalAddress { get; private set; }
 
         public Address RemoteAddress { get; private set; }
+
+        public DisassociateInfo? DisassociationInfo { get; private set; }
     }
 
     /// <summary>
     /// INTERNAL API
     /// </summary>
-    internal sealed class HopelessAssociationException : EndpointException, IAssociationProblem
+    internal sealed class HopelessAssociation : EndpointException, IAssociationProblem
```

Comment:
Changed the name of this exception to match the JVM (addressed some earlier comments by @rogeralsing)